core/src/main/java/io/questdb/log/LogRollingFileWriter.java

```diff
@@ -244,6 +278,40 @@ private void flush() {
         _wptr = buf;
     }
 
+    private void removeOldLogs(){
+        if(this.lifeDuration != null) {
+            ff.iterateDir(path.of(logDir).$(), removeExpiredLogsVisitor);
+        }
+        if(this.sizeLimit != null) {
+            currentLogSizeSum = 0;
+            ff.iterateDir(path.of(logDir).$(), removeExcessiveLogsVisitor);
+        }
+    }
+    private void removeOldLogsVisitor(long filePointer, int type){
+        path.trimTo(logDir.length());
+        path.concat(filePointer).$();
+        NativeLPSZ name = new NativeLPSZ();
+        name.of(filePointer);
+        if (Files.notDots(filePointer) && type == Files.DT_FILE && name.toString().contains(logFileTemplate)
```

Comment:
Please use `Chars#contains()` instead of `name.toString().contains()`. We tend to avoid generating litter where possible.

The same applies to `removeExcessiveLogsVisitor`.